x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -252,10 +281,14 @@ export interface ActionDetails {
    * The list of action log items (actions and responses) received thus far for the action.
    */
   logEntries: ActivityLogEntry[];
+  /**
+   * The output data from an action
+   */
+  outputs?: OutputsType<TOutputType>;
```

Comment:
If you decide to go with my suggestion above, then this here would be:

```typescript
export interface ActionDetails<TOutputContent extends object = object> {
  //...

  /** The output from each Endpoint to the action */
  outputs: Record<string,  ActionResponseOutput<TOutputContent>>

  //...
```

Note: I'm suggesting the `outputs` not be optional here and if the action does not return any `output`, then it can be set to an empty object on the response `{}`.